source/common/http/codec_client.cc

```diff
@@ -128,8 +128,25 @@ void CodecClient::responsePreDecodeComplete(ActiveRequest& request) {
   if (codec_client_callbacks_) {
     codec_client_callbacks_->onStreamPreDecodeComplete();
   }
-  deleteRequest(request);
 
+  request.decode_complete_ = true;
```

Comment:
if the DecodeComplete calls can handle the decode completion, should we use RequestEncoderWrapper/onEncodeComplete for the encode  side?

Reply:
Oh, yeah, that looks like a good idea.